homeassistant/components/tuya/fan.py

```diff
@@ -137,7 +137,7 @@ def set_percentage(self, percentage: int) -> None:
                 [
                     {
                         "code": self._speed.dpcode,
-                        "value": int(self._speed.remap_value_from(percentage, 0, 100)),
+                        "value": int(self._speed.remap_value_from(percentage, 1, 100)),
```

Comment:
This PR contains to different things. One is a bugfix, the other a new feature.

These should be separated into separate PRs, as the bugfix could go into the current beta, the new feature however, cannot.